File: code/reviewServer/src/main/java/com/jay/reviewserver/domain/entity/InternalImage.java
package com.jay.reviewserver.domain.entity;

import jakarta.persistence.Access;
import jakarta.persistence.AccessType;
import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;

@Entity
@DiscriminatorValue("II")
@Access(AccessType.FIELD)
public class InternalImage extends ReviewImage {
    @Override
    public String url() {
        return "/images/original/" + getPath();
    }

    @Override
    public Boolean hasThumbnail() {
        return true;
    }

    @Override
    public String getThumbnail() {
        return "/images/original/" + getPath();
    }
}

File: code/productserver/src/main/kotlin/com/jay/productserver/domain/entity/InternalImage.kt
package com.jay.productserver.domain.entity

import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity

@Entity
@DiscriminatorValue("II")
class InternalImage (
    path : String
) : ProductImage(path = path) {
    override fun url(): String {
        return "/images/original/" + getPath()
    }

    override fun hasThumbnail(): Boolean {
        return true
    }

    override fun thumbnailURL(): String {
        return "images/thumbnail/" + getPath()
    }
}